// Protocol buffer for graph to sequence encoding.
//
// Copyright 2019 the ProGraML authors.
//
// Contact Chris Cummins <chrisc.101@gmail.com>.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//    http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto2";

package ml4pl;

import "deeplearning/ml4pl/graphs/programl.proto";

// An graph that has been serialized and encoded to a concatenated sequence of
// encoded node representations.
message ProgramGraphSeq {
  // The concatenated sequences of encoded node representations.
  repeated int32 encoded = 1;
  repeated int32 encoded_node_length = 2;
  // A list of nodes in the graph, in the range [0, len(graph.node)].
  repeated int32 node = 3;
}

// A GraphEncoder job, containing a vocabulary, a list of graphs to encode,
// and the encoded results.
message GraphEncoderJob {
  map<string, int32> vocabulary = 1;
  repeated ProgramGraph graph = 2;
  repeated ProgramGraphSeq seq = 3;
}
